src/azul/plugins/repository/tdr/__init__.py

```diff
@@ -176,6 +182,38 @@ def create(cls, catalog: CatalogName) -> 'RepositoryPlugin':
     def sources(self) -> AbstractSet[TDRSourceSpec]:
         return self._sources
 
+    @cached_property
+    def _configured_source_specs_by_name(self) -> Dict[str, TDRSourceSpec]:
+        catalog_sources = itertools.chain.from_iterable(
+            config.tdr_sources(catalog)
+            for catalog in config.catalogs
+        )
+        return {
+            source.name: source
+            for source in map(TDRSourceSpec.parse, catalog_sources)
+        }
+
+    def list_sources(self,
+                     authentication: Optional[Authentication]
+                     ) -> List[TDRSourceRef]:
+        if isinstance(authentication, OAuth2):
+            tdr = TDRClient.with_user_credentials(authentication)
+            snapshot_ids_by_name = tdr.snapshot_ids_by_name()
```

Comment:
Same concern here. While snapshot names are de-facto unique at the moment and the DCP/2 spec requires them to be, the called method does not only return DCP/2 catalogs so I wouldn't want to rely on names in that list being unique. In this case it's easy to invert the mapping.